server/command.go

```diff
@@ -249,6 +251,8 @@ func (p *Plugin) executeUnlinkCommand(args *model.CommandArgs) (*model.CommandRe
 
 	p.sendBotEphemeralPost(args.UserId, args.ChannelId, "The MS Teams channel is no longer linked to this Mattermost channel.")
 
+	p.storeMutex.Lock()
```

Comment:
Why this `storeMutex` is needed? it looks like is going to generate a ton of contention.

Reply:
We added this to avoid concurrent reading and writing in the database. Let's say we add a reaction on Mattermost, and in the plugin hook we make an API call to add the corresponding reaction on MS Teams. Before storing the `lastUpdateAt` time in the database, we receive an update event from MS Teams, and we check for the `lastUpdateAt` time in the database for the corresponding post which might not be updated at that moment due to which we again make changes on Mattermost. To avoid these conditions we apply a mutex lock.

Reply:
Databases are for handling concurrent reading and writting of the data, having this kind of database access is going to lead to a huge bottleneck in the database access.

My suggestion would be handle that case with transactions instead, but in reality, what I would expect here is to have certain degree of eventual consistency, I prefer to have some extra notifications there to handle there than this huge bottleneck.

Clearly we have to find a better way to solve the same problem, this is not something that we can deploy into production.